test/Datasource.test.js

```diff
@@ -1,116 +1,190 @@
 'use strict';
 const expect = require('expect');
 const fs = require('fs');
-const FileReadWriteDatasource = require('../src/lib/datasource/FileReadWriteDatasource');
-
+// const FileReadWriteDataSource = require('../src/lib/datasource/FileReadWriteDataSource');
+const inMemoryDataSource = require('../src/lib/datasource/InMemoryDataSource');
 let dataSource, brokenDataSource, nonExistantDataSource, emptyDataSource;
+let fixtureObject = {
+  model: 'IPhone',
+  brand: 'Apple'
+}
 
-describe.skip('Test all datasouses from ../src/lib/datasouce', () => {
-    before(() => {
-        //  todo: review this, and add multiply classes structure
-        const filename = './fixtures/dataCorrect.json';
-        dataSource = new FileReadWriteDatasource(filename);
-    
-        const brokenFilename = './fixtures/dataIncorrect.doc';
-        brokenDataSource = new FileReadWriteDatasource(brokenFilename);
-    
-        const nonExistantFilename = './Test/fixtures/dataCorrect.doc'    
-        nonExistantDataSource = new FileReadWriteDatasource(nonExistantFilename);
+describe('Test all datasouses from ../src/lib/datasouce', () => {
+  before(() => {
+      dataSource = new inMemoryDataSource();
+
```

Comment:
Need new code for testing all "datasource" classes at a time.

Reply:
k. going to add it soon

Reply:
In a meantime, consider to implement an abstract function that launches tests over some configurable datasource just as we discussed